api/src/controllers/reviews.ts

```diff
@@ -3,185 +3,162 @@
 */
 
 import { verifyCaptcha } from '../helpers/recaptcha';
-import Review from '../models/review';
-import Vote from '../models/vote';
-import Report from '../models/report';
-import mongoose from 'mongoose';
 import { adminProcedure, publicProcedure, router, userProcedure } from '../helpers/trpc';
 import { z } from 'zod';
-import { editReviewSubmission, featuredQuery, ReviewData, reviewSubmission } from '@peterportal/types';
+import {
+  anonymousName,
+  editReviewSubmission,
+  featuredQuery,
+  FeaturedReviewData,
+  ReviewData,
+  reviewSubmission,
+} from '@peterportal/types';
 import { TRPCError } from '@trpc/server';
+import { db } from '../db';
+import { review, user, vote } from '../db/schema';
+import { and, count, desc, eq, sql } from 'drizzle-orm';
 
-interface ReviewFilter {
-  courseID?: string;
-  professorID?: string;
-  userID?: string;
-  _id?: mongoose.Types.ObjectId;
-  verified?: boolean;
-}
+async function userWroteReview(userId: number | undefined, reviewId: number) {
+  if (!userId) {
+    return false;
+  }
 
-interface VoteData {
-  _id?: string;
-  userID: string;
-  reviewID: string;
-  score: number;
+  return (
+    (
+      await db
+        .select({ count: count() })
+        .from(review)
+        .where(and(eq(review.id, reviewId), eq(review.userId, userId)))
+    )[0].count > 0
+  );
 }
 
-async function userWroteReview(userID: string | undefined, reviewID: string) {
-  if (!userID) {
-    return false;
+async function getReviews(
+  where: {
+    courseId?: string;
+    professorId?: string;
+    userId?: number;
+    reviewId?: number;
+    verified?: boolean;
+  },
+  sessUserId?: number,
+) {
+  const { courseId, professorId, userId, reviewId, verified } = where;
+  const userVoteSubquery = db
+    .select({ reviewId: vote.reviewId, userVote: vote.vote })
+    .from(vote)
+    .where(eq(vote.userId, sessUserId!))
+    .as('user_vote_query');
+  const results = await db
+    .select({
+      review: review,
+      score: sql`COALESCE(SUM(${vote.vote}), 0)`.mapWith(Number),
+      userDisplay: user.name,
+      userVote: sql`COALESCE(${userVoteSubquery.userVote}, 0)`.mapWith(Number),
+    })
+    .from(review)
+    .where(
+      and(
+        ...[
+          ...(courseId ? [eq(review.courseId, courseId)] : []),
+          ...(professorId ? [eq(review.professorId, professorId)] : []),
+          ...(userId ? [eq(review.userId, userId)] : []),
+          ...(reviewId ? [eq(review.id, reviewId)] : []),
+          ...(verified !== undefined ? [eq(review.verified, verified)] : []),
+        ],
```

Comment:
This seems a bit clunky. At the very least we don't have to construct an array just to spread it. If anything, I'd prefer to declare `const filters = []` and simply add each `eq()` to the filters array, then just do `.and(...filters)` at the end.

```suggestion
        ...(courseId ? [eq(review.courseId, courseId)] : []),
        ...(professorId ? [eq(review.professorId, professorId)] : []),
        ...(userId ? [eq(review.userId, userId)] : []),
        ...(reviewId ? [eq(review.id, reviewId)] : []),
        ...(verified !== undefined ? [eq(review.verified, verified)] : []),
```

Reply:
Yes, I'll remove the outer array. I'm really not sure if adding will look any cleaner though since we would just be adding based on these same conditions so it would look kind of the same. I took the spreading idea from [drizzle docs](https://orm.drizzle.team/docs/select#conditional-select).

Reply:
Ok I found another reference on the [drizzle docs](https://orm.drizzle.team/docs/guides/conditional-filters-in-query) and I was able to remove the spread operators.